app/controllers/api/v1/extensions_controller.rb

```diff
@@ -110,4 +113,34 @@ def index
   def show
     @extension = Extension.with_owner_and_lowercase_name(owner_name: params[:username], lowercase_name: params[:id])
   end
+
+  api! <<~EOD
+    Recompile #{I18n.t('nouns.extension').pluralize} in the #{Rails.configuration.app_name}.
+    This endpoint functions for authenticated users (via API auth), and authorized users (site owner, admin).
+    Requires a header tag of X-Ops-Userid with the username of the authenticated user.
+    Responds with either success or error.
+  EOD
+  param :username, String, required: true, desc: "#{Rails.configuration.app_name} user name of the #{I18n.t('nouns.extension')} owner"
+  param :id,       String, required: true, desc: "#{Rails.configuration.app_name} #{I18n.t('nouns.extension')} name"
+  example "GET https://#{ENV['HOST']}/api/v1/assets/recompile/demillir/maruku/"
+  example "{'message'=>'Please wait a minute or so for the repository releases to be recompiled, then refresh this page to see the updates.'}"
+
+  def sync_repo
+    extension = Extension.with_owner_and_lowercase_name(owner_name: params[:username], lowercase_name: params[:id])
+    if extension.blank?
+      render json: {error_code: I18n.t('api.error_codes.not_found'), error_messages: [I18n.t('api.error_messages.not_found')] }, status: 404
+      return
+    end
+    begin
+      authorize! extension unless current_user.is?(:admin)
+    rescue
+      render_not_authorized([t('api.error_messages.unauthorized_recompile_error')])
+    else
+      SyncExtensionRepoWorker.perform_async(extension.id)
```

Comment:
The `SyncExtensionRepoWorker` only works for GitHub-based extensions.  We need this new API endpoint to work for hosted extensions, too.

Reply:
To make this code work for hosted- and GitHub-based extensions, you can call the new `CompileExtension` service class in #86.